### ISSUE TITLE
can't display date properties

### ISSUE THREAD

**Issue by macrozone** [GriddleGriddle/Griddle#613]
## Griddle version

Version 1.1.0

## Expected Behavior

cell that contains a property with Date-type should display property

## Actual Behavior

the cell is empty, also if you specify customComponent `value` is an empty Map object instead of the date,.

## Steps to reproduce

```
<Griddle
  date={[
     { foo: "row1", date: new Date() },
     { foo: "row2", date: new Date() },
]}
/>
```

**Comment by macrozone** [comment]
I tried to debug this. Problem seems to be the conversion to Immutable.js:

https://github.com/GriddleGriddle/Griddle/blob/22901b3402ac29833ff89983d81e69b9745a67af/src/utils/dataUtils.js#L18

The Map does not preserve things that are not objects or Arrays as it seems, so a date just gets converted to an empty object (!)

can we somehow use BEFORE_REDUCE or so to transform dates to numbers? (timestamp)

**Comment by macrozone** [comment]
I tried to create a plugin that transforms the data, but i had no luck. I tried GRIDDLE_LOADED_DATA with and without _BEFORE, BEFORE_REDUCE, but I could not find the right hook to transform the data. It's super confusing.

Having Date-objects in the data-grid is such a common scenario that i don't know what i am doing wrong.

**Comment by ryanlanciaux** [comment]
Sorry you're bumping into this. If you could, please submit a storybook story with the steps you mentioned above. The steps are pretty straightforward and normally I would do this but pretty swamped this week so that would be incredibly helpful. 

Thanks!

**Comment by macrozone** [comment]
good idea,
here we are: https://github.com/GriddleGriddle/Griddle/pull/614

**Comment by ryanlanciaux** [comment]
Thanks a ton!

**Comment by joellanciaux** [comment]
Thanks a ton for putting together this issue + the _very_ helpful story PR :+1:

I've made two updates that should make displaying dates a little easier:
- As you noted, there was an issue in `dataUtil` that was blowing away the `date` values. I've added an additional condition so that `dates` are preserved in the same manner as Immutable's default `fromJS`.
- I've made an update so that one can specify a `type` on the column definition to facilitate automatic conversion. Ex: `<ColumnDefinition ... type='date' />`

This will be available in `Griddle@1.2.0`, which should be released pretty quick.

Thanks again!
